clients/src/main/resources/common/message/SaslHandshakeResponse.json

```diff
@@ -18,9 +18,9 @@
   "type": "response",
   "name": "SaslHandshakeResponse",
   // Version 1 is the same as version 0.
-  // Version 2 adds flexible version support
-  "validVersions": "0-2",
-  "flexibleVersions": "2+",
+  // NOTE: Version cannot be easily bumped due to incorrect
+  // client negotiation. See https://issues.apache.org/jira/browse/KAFKA-9577
```

Comment:
Can you specify the client versions with the incorrect negotiation?  2.4 and earlier, right?

Reply:
That's right.

Reply:
Updated.